### PR TITLE
Merge 1.5 to 1.6

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

Merge 1.5 into 1.6

What does this Pull Request do? reference the related issue?

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added Behats                      | n
| Changelog updated                 | n
| Review and 2 GTM                  | n
| Micro Demo to the PO (Story only) | n
| Migration script                  | n
| Tech Doc                          | n

### CODE REVIEW

File: web/app.php [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-$loader->register(true);
-*/
-
 require_once __DIR__.'/../app/AppKernel.php';
 //require_once __DIR__.'/../app/AppCache.php';
 
 $kernel = new AppKernel('prod', false);
 $kernel->loadClassCache();
 //$kernel = new AppCache($kernel);
+
```
Comment (solivier) [inline web/app.php]: be careful on next tag it also has to be on the standard edition

File: CHANGELOG-1.5.md [hunk 2]
```diff
@@ -1,8 +1,21 @@
-# 1.5.x
+# 1.5.9 (2016-09-27)
+
+## Bug fixes
+clear
```
Comment (solivier) [inline CHANGELOG-1.5.md]: to remove

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/IndexCreator.php [hunk 3]
```diff
@@ -121,7 +121,7 @@ public function ensureIndexesFromChannel(ChannelInterface $channel)
      *
      * @param LocaleInterface $locale
      */
-    public function ensureIndexesFromLocale(LocaleInterface $locale)
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/IndexCreator.php]: BC ?

File: features/Context/Page/Base/Form.php [hunk 4]
```diff
@@ -64,6 +64,87 @@ public function saveAndClose()
...
+    }
+
+    /**
+     * Get the specified tab
+     *
+     * @return NodeElement
+     */
+    public function getTab($tab)
+    {
+        return $this->find('css', sprintf('a:contains("%s")', $tab));
```
Comment (grena) [inline features/Context/Page/Base/Form.php]: spin

File: features/product/filtering/filter_products_per_with_multiple_multiselect.feature [hunk 5]
```diff
@@ -30,6 +30,7 @@ Feature: Filter products with multiples multiselect filters
     And I am logged in as "Mary"
     And I am on the products page
 
+  @unstable
```
Comment (grena) [inline features/product/filtering/filter_products_per_with_multiple_multiselect.feature]: check on 1.6

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/IndexCreator.php (hunk 3 above)
Comment (grena) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/IndexCreator.php]: Ok, so no BC.

File: src/Pim/Bundle/ConnectorBundle/Resources/translations/messages.pt.yml [hunk 6]
```diff
@@ -169,138 +220,3 @@ job_execution.summary:
...
-  xlsx_channel_import:
-    validation.label: Validação do ficheiro
-  xlsx_locale_import:
-    validation.label: Validação do ficheiro
-  xlsx_attribute_group_import:
-    validation.label: Validação do ficheiro
-  xlsx_currency_import:
-    validation.label: Validação do ficheiro
-  xlsx_group_type_import:
-    validation.label: Validação do ficheiro
```
Comment (grena) [inline src/Pim/Bundle/ConnectorBundle/Resources/translations/messages.pt.yml]: re-run nelson after merge

File: src/Pim/Bundle/EnrichBundle/Resources/config/controllers.yml [hunk 7]
```diff
@@ -457,8 +459,14 @@ services:
             - '@pim_catalog.builder.product_template'
 
     pim_enrich.controller.rest.versioning:
-        class: '%pim_enrich.controller.rest.versioning.class%'
+        class: %pim_enrich.controller.rest.versioning.class%
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/config/controllers.yml]: keep the quotes

File: src/Pim/Bundle/EnrichBundle/Resources/config/controllers.yml [hunk 8]
```diff
@@ -457,8 +459,14 @@ services:
...
         arguments:
             - '@pim_versioning.repository.version'
             - '@pim_catalog.resolver.fqcn'
             - '@pim_internal_api_serializer'
+
+    pim_enrich.controller.rest.attribute_option:
+        class: %pim_enrich.controller.rest.attribute_option.class%
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/config/controllers.yml]: quotes

File: src/Pim/Bundle/EnrichBundle/Resources/config/normalizers.yml [hunk 9]
```diff
@@ -144,3 +145,12 @@ services:
         class: %pim_enrich.normalizer.variant_group_violation.class%
         arguments:
             - '@pim_internal_api_serializer'
+
+    pim_enrich.normalizer.structured.attribute_option:
+        class: %pim_enrich.normalizer.structured.attribute_option.class%
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/config/normalizers.yml]: quotes maggle

File: src/Pim/Bundle/EnrichBundle/Resources/config/repositories.yml [hunk 10]
```diff
@@ -76,3 +77,9 @@ services:
             - '@pim_user.context.user'
             - '@doctrine.orm.entity_manager'
             - '%pim_catalog.entity.attribute_group.class%'
+
+    pim_enrich.repository.attribute_option.search:
+        class: %pim_enrich.repository.attribute_option.search.class%
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/config/repositories.yml]: quotes

File: src/Pim/Bundle/JsFormValidationBundle/Generator/FormValidationScriptGenerator.php [hunk 11]
```diff
@@ -139,18 +139,19 @@ protected function generateValidationParameters(FormView $formView)
             $dispatcher->dispatch(JsfvEvents::preProcess, $preProcessEvent);
             // @codingStandardsIgnoreEnd
 
-            if (!empty($metadata->constraints)) {
-                foreach ($metadata->constraints as $constraint) {
+            if ($metadata->hasConstraints()) {
+                foreach ($metadata->getConstraints() as $constraint) {
                     $constraintName = $this->getConstraintName($constraint);
-                    if ($constraintName == 'UniqueEntity') {
+                    if ('UniqueEntity' == $constraintName) {
```
Comment (grena) [inline src/Pim/Bundle/JsFormValidationBundle/Generator/FormValidationScriptGenerator.php]: triple

File: src/Pim/Bundle/JsFormValidationBundle/Generator/FormValidationScriptGenerator.php [hunk 12]
```diff
@@ -139,18 +139,19 @@ protected function generateValidationParameters(FormView $formView)
...
                         if (is_array($constraint->fields)) {
                             //It has not been implemented yet
                         } elseif (is_string($constraint->fields)) {
-                            if (!isset($aConstraints[$constraint->fields])) {
-                                $aConstraints[$constraint->fields] = [];
-                            }
                             $aConstraints[$constraint->fields][] = $constraint;
                         }
+                    } elseif ('NotBlankProperties' == $constraintName) {
```
Comment (grena) [inline src/Pim/Bundle/JsFormValidationBundle/Generator/FormValidationScriptGenerator.php]: triple

**Comment by grena** [comment]
Mob reviewed by @skeleton , @solivier , @juliensnz & myself.

**Pull request merged**